src/stylesheets/elements/form-controls/_checkbox-and-radio.scss

```diff
@@ -215,18 +218,18 @@
   margin-top: units(1.5);
   padding-left: units($input-select-margin-right + $theme-input-select-size);
   position: relative;
-  text-indent: units(-$input-select-margin-right - $theme-input-select-size);
 
   &:before {
     content: " ";
-    display: inline-block;
-    left: units($theme-input-select-border-width);
-    line-height: units($theme-input-select-size);
-    margin-right: units($input-select-margin-right);
-    position: relative;
-    text-indent: 0;
-    vertical-align: middle\0; // Target IE 11 and below to vertically center inputs
-    white-space: pre;
+    display: block;
+    left: 0;
+    margin-left: units($theme-input-select-border-width);
+    margin-top: (
+      line-height($theme-form-font-family, $theme-input-line-height) *
+      font-size($theme-form-font-family, $theme-body-font-size) -
```

Comment:
This may need a rethink, since it assumes font styles inherited from `.usa-fieldset`, which are likely not guaranteed.

https://github.com/uswds/uswds/blob/f3bdc643dac135ef51ae75e91cde96b0a4d8c7b3/src/stylesheets/elements/form-controls/_global.scss#L32-L35

Reply:
This is a situation where it would be nice to have the [CSS4 `lh` unit](https://www.w3.org/TR/css-values-4/#font-relative-lengths), since essentially what we want is `margin-top: calc((1lh - #{units($theme-input-select-size)}) / 2);`.

A couple alternatives to consider:

- Maybe the checkbox should explicitly set font expectations, which might be desirable anyways. i.e. Add the `%block-input-general` placeholder to `usa-checkbox`.
- Another I'd hoped could work is to chop up the elements so that the natural height of the label is only the label text excluding the description, so that a standard vertical centering approach could work. I've got this close to working in a [CodePen prototype](https://codepen.io/aduth/pen/GRmzQaO), but positioning the description absolutely means that it won't "push down" content that occurs after the checkbox.

Reply:
@aduth have you tried Flexbox? Seems similar to the [media object pattern [solved by flexbox - philipwalton.github.io]](https://philipwalton.github.io/solved-by-flexbox/demos/media-object/)

Reply:
> * Maybe the checkbox should explicitly set font expectations, which might be desirable anyways. i.e. Add the `%block-input-general` placeholder to `usa-checkbox`.

I went this direction in 9e958ae, based on there being some precedent of this applying at the component level [[1]](https://github.com/uswds/uswds/blob/6218825ad5b5054057bc8d3dd0f7255296ddeaa9/src/stylesheets/elements/form-controls/_combo-box.scss#L21)[[2]](https://github.com/uswds/uswds/blob/6218825ad5b5054057bc8d3dd0f7255296ddeaa9/src/stylesheets/elements/form-controls/_dropdown.scss#L2)[[3]](https://github.com/uswds/uswds/blob/6218825ad5b5054057bc8d3dd0f7255296ddeaa9/src/stylesheets/elements/form-controls/_range-input.scss#L30)[[4]](https://github.com/uswds/uswds/blob/6218825ad5b5054057bc8d3dd0f7255296ddeaa9/src/stylesheets/elements/form-controls/_text-input.scss#L3), where presumably it's expected that the checkbox and radio could (or should) inherit from the input-related typography settings.